core/runtime/src/main/java/io/quarkus/runtime/configuration/ProfileManager.java

```diff
@@ -40,6 +40,7 @@ public static void setRuntimeDefaultProfile(final String profile) {
     }
 
     //NOTE: changes made here must be replicated in BootstrapProfile
+    @Deprecated
```

Comment:
Add a Javadoc explaining why this is deprecated and what's the alternative?

Reply:
Added javadoc.